openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -228,6 +232,16 @@ def from_component(cls, library_key, component):
         """
         last_publish_log = component.versioning.last_publish_log
 
+        published_by = None
+        if last_publish_log and last_publish_log.published_by:
+            published_by = last_publish_log.published_by.username
+
+        last_draft_log = authoring_api.get_entities_with_unpublished_changes(component.pk) \
+            .order_by('-created').first()
+        last_draft_created = last_draft_log.created if last_draft_log else None
+        last_draft_created_by = \
+            last_draft_log.created_by.username if last_draft_log and last_draft_log.created_by else None
```

Comment:
We can access this info from [`component.versioning.draft`](https://github.com/openedx/openedx-learning/blob/6fa83ba94c32fa6a1d2abddfe666ffcb0c628a36/openedx_learning/apps/authoring/publishing/model_mixins.py#L152-L179), which will always point to the last draft version, if there is one. (If the component's latest version is published, then `component.versioning.draft == component.versioning.published`, but `draft` still points to the last draft version.)

Unfortunately we [only have the `created` property exposed in openedx-learning](https://github.com/openedx/openedx-learning/blob/6fa83ba94c32fa6a1d2abddfe666ffcb0c628a36/openedx_learning/apps/authoring/publishing/model_mixins.py#L307-L309), but a minor update would fix that if you have time to submit it?

```suggestion
        draft = component.versioning.draft
        last_draft_created = draft.created if draft else None
        last_draft_created_by = draft.publishable_entity_version.created_by if draft else None
```

Reply:
Created a PR here: https://github.com/openedx/openedx-learning/pull/227